util/oidc/oidc.go

```diff
@@ -367,12 +392,45 @@ func (a *ClientApp) HandleCallback(w http.ResponseWriter, r *http.Request) {
 	if a.secureCookie {
 		flags = append(flags, "Secure")
 	}
-	var claims jwt.MapClaims
+	claims := Claims{}
 	err = idToken.Claims(&claims)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	groups := claims.Groups
+
+	// Some SSO implementations (Okta) require a call to
+	// the OIDC user info path to get attributes like groups
+	if a.userInfoPath != "" {
+		groups, err = a.GetUserInfoGroups(token.AccessToken, a.settings.IssuerURL(), a.userInfoPath)
+		if err != nil {
+			w.WriteHeader(401)
+			_, _ = w.Write([]byte(fmt.Sprintf("failed to get groups claim: %v", err)))
+			return
+		}
+	}
+
+	// reassamble the JWT claims
+	argoClaims := Claims{
+		groups,
+		jwt.RegisteredClaims{
+			ExpiresAt: jwt.NewNumericDate(time.Now().Add(a.settings.UserSessionDuration)),
+			IssuedAt:  jwt.NewNumericDate(time.Now()),
+			NotBefore: jwt.NewNumericDate(time.Now()),
+			Issuer:    idToken.Issuer,
+			Subject:   idToken.Subject,
+			Audience:  idToken.Audience,
+		},
+	}
+	newToken := jwt.NewWithClaims(jwt.SigningMethodRS256, argoClaims)
+	idTokenRAW, err = newToken.SignedString(a.encryptionKey)
+	if err != nil {
+		w.WriteHeader(401)
```

Comment:
Ditto above, this error seems more like a 5XX than a 401.

Reply:
You're right, changed it to 500